### ISSUE TITLE
[Suggestion] Each post should include share buttons

### ISSUE THREAD

**Issue by nisrulz** [zzossig/hugo-theme-zzo#97]
Being able to share posts to other media platforms will fall inline with blog's use case.

Here is an example:
![Screenshot from 2020-01-20 21-34-10](https://user-images.githubusercontent.com/2096087/72756187-c0aaaf00-3bcc-11ea-8c8c-c49fad3c0736.png)

You can use the below shield.io badges also:

Code:
```html
<a href="https://twitter.com/intent/tweet?text=EasyDeviceInfo%3A%20Android%20library%20to%20get%20device%20information%20in%20a%20super%20easy%20way%F0%9F%98%8E&url=https://github.com/nisrulz/easydeviceinfo&via=nisrulz&hashtags=AndroidDev">
        <img src="https://img.shields.io/twitter/url/http/shields.io.svg?style=social"/>
    </a></h1>
```
Preview: <a href="https://twitter.com/intent/tweet?text=EasyDeviceInfo%3A%20Android%20library%20to%20get%20device%20information%20in%20a%20super%20easy%20way%F0%9F%98%8E&url=https://github.com/nisrulz/easydeviceinfo&via=nisrulz&hashtags=AndroidDev">
        <img src="https://img.shields.io/twitter/url/http/shields.io.svg?style=social"/>
    </a></h1>

> Note: "EasyDeviceInfo%3A%20Android%20library%20to%20get%20device%20information%20in%20a%20super%20easy%20way%F0%9F%98%8E" is a URL encoded string. This can be the title of the blog post but you would need to url encode the title before appending here.

**Comment by nisrulz** [comment]
Btw, I moved my blog completely to your theme and it is active now: https://crushingcode.nisrulz.com/

**Comment by zzossig** [comment]
Looks good your blog, I'll keep watching it.

I was thinking about the share button. I'll add the feature

**Comment by zzossig** [comment]
Do you need a specific share button?
I did `Facebook` and `twitter` but there are a lot of other things.

**Comment by nisrulz** [comment]
I was thinking of Reddit too :thinking:

**Comment by zzossig** [comment]
Oh, thanks I’ll add it tomorrow:)

**Comment by zzossig** [comment]
Your Reddit code works Thx!
Now we have social options

params.toml file
```
...
[share]
  options = ["facebook","twitter", "reddit", "linkedin", "tumblr", "weibo", "douban", "line"]
```

**Comment by nisrulz** [comment]
Nice!

Few observations:

For twitter, only the title is getting populated in the share box. I noticed you don't add this in the link `&url=https://github.com/nisrulz/easydeviceinfo&via=nisrulz&hashtags=AndroidDev` 

So it shows up like this:

<img width="681" alt="Screen Shot 2020-01-23 at 9 43 28 AM" src="https://user-images.githubusercontent.com/2096087/72968994-f1553900-3dc4-11ea-8a8b-5c50efd4b598.png">

The link can take more param:
- url
- via
- hastags

If you did add those params as I shared in the description of this issue, your share link would work like below:

<img width="681" alt="Screen Shot 2020-01-23 at 9 43 53 AM" src="https://user-images.githubusercontent.com/2096087/72969108-25305e80-3dc5-11ea-8585-04520ec74491.png">

For the theme, the 
- hastags would be same as the tags specified in the page. If not specified then don't add this param.
- via would be same as twitter username specified. If not specified then don't add this param.
- url will be the url of the page.

This kind of functionality exists in almost all kind of share url. I can get you the code for those.

Reddit I already shared.
Twitter I just explained.

Facebook and others I can find.

**Comment by nisrulz** [comment]
Also this text should be `Tumbler`

<img width="98" alt="Screen Shot 2020-01-23 at 9 48 44 AM" src="https://user-images.githubusercontent.com/2096087/72969363-9839d500-3dc5-11ea-82fa-954aa01ee64b.png">

**Comment by zzossig** [comment]
Thx for testing it again~! I appreciate!
Ok, I didn't know that the twitter parameters.
I'll add the feature.

**Comment by zzossig** [comment]
I added the hashtags, via, url to the twitter share button

**Comment by nisrulz** [comment]
Did you change something in the layout of share buttons. I just updated the git sub module and I got this:

![Screenshot from 2020-01-23 19-25-10](https://user-images.githubusercontent.com/2096087/73012203-1f169e00-3e16-11ea-816d-8997bee9629d.png)

**Comment by zzossig** [comment]
Ops, I’ll change back

**Comment by nisrulz** [comment]
Also, I think you can do away with the outline on the button and the text. Social media is icons are pretty much known to every, so just the logo should be more elegant.

**Comment by zzossig** [comment]
The share button layout works well on my end. Can you force refresh the browser?

**Comment by zzossig** [comment]
Did you mean this?
![주석 2020-01-24 034048](https://user-images.githubusercontent.com/52706977/73013265-52c7e700-3e5b-11ea-86d9-22c49bd09796.png)

**Comment by nisrulz** [comment]
Everything work now :taco: 

Thank you!

**Comment by zzossig** [comment]
But, I have a plan to redesign the share buttons. It looks like awkward now
![주석 2020-01-24 161745](https://user-images.githubusercontent.com/52706977/73051062-23e65b00-3ec5-11ea-8fc5-eb7022df4882.png)

**Comment by nisrulz** [comment]
Most blogs I read have them like this:

<img width="211" alt="Screen Shot 2020-01-24 at 8 39 28 AM" src="https://user-images.githubusercontent.com/2096087/73052341-3ccf1c00-3e85-11ea-972b-977eecc411f0.png">
<img width="159" alt="Screen Shot 2020-01-24 at 8 39 53 AM" src="https://user-images.githubusercontent.com/2096087/73052342-3d67b280-3e85-11ea-8106-9605d037817e.png">
<img width="215" alt="Screen Shot 2020-01-24 at 8 40 24 AM" src="https://user-images.githubusercontent.com/2096087/73052343-3d67b280-3e85-11ea-88af-5f60f162b272.png">

**Comment by zzossig** [comment]
The Facebook share button never worked on the localhost.
I checked that the URL is accurately connected to the button.

When you decode the URL, 
![주석 2020-01-25 025724](https://user-images.githubusercontent.com/52706977/73091910-caab1580-3f1e-11ea-9319-6506f8d51d58.png)
it gives me this text

```
href="https://www.facebook.com/sharer/sharer.php?u=http://localhost:3796/posts/javascript/ball_with_mouse/"
```

And this is the right URL.
Just the localhost is the problem.

And I changed the button design😋
